### PR TITLE
Logout should only clear local cache when run inside iframe

### LINKED ISSUE #2145: How to disable post-logout redirect
Please follow the issue template below. Failure to do so will result in a delay in answering your question.

## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.0.0`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description
The default logout behavior redirects users back to the current location.href. This doesn't work for my app because if a user is not authenticated we auto redirect users to login creating a confusing UX on logout. Instead, I would like to not redirect the user at all after the logout is successful. How can I disable the default redirect behavior? I've tried setting the msal config `postLogoutRedirectUri: ''` without success. 
I believe our current auth solution achieves this behavior by simply not providing the redirect param.

### PR DESCRIPTION
**Pull request by jasonnutter**
When an application's logout broadcast url is loaded (known as "Logout URL" in the portal, and is different than the post logout redirect uri), we should be advising applications to invoke the MSAL logout function to clear local cache. While this can be done today, the logout function should not attempt to redirect, as the logout broadcast url will be loaded inside an iframe.

This PR will also address the bugs raised in:
- #2452
- #2145 

The application's logout broadcast URL will be given the session `sid` and `iss` as query parameters. Guidance from the server is to inspect cached accounts and if there is an account that matches the provided `sid`, only remove that account (this will require exposing `idTokenClaims` in `AccountInfo`). If no account matches the given `sid` (likely because `sid` was not enabled for the app), then clear all accounts. 

TODO:
- [ ] Tests
- [ ] Documenation about logout and logout broadcast url

### CODE REVIEW

File: lib/msal-common/src/client/AuthorizationCodeClient.ts [hunk 1]
```diff
@@ -105,24 +104,16 @@ export class AuthorizationCodeClient extends BaseClient {
             throw ClientConfigurationError.createEmptyLogoutRequestError();
         }
 
-        if (logoutRequest.account) {
-            // Clear given account.
-            this.cacheManager.removeAccount(AccountEntity.generateAccountCacheKey(logoutRequest.account));
-        } else {
-            // Clear all accounts and tokens
-            this.cacheManager.clear();
-        }
```
Comment (jasonnutter) [inline lib/msal-common/src/client/AuthorizationCodeClient.ts:114]: Clearing cache and creating logout url should be separated.

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 2]
```diff
@@ -554,12 +554,33 @@ export abstract class ClientApplication {
...
+             */
+            if (logoutRequest && logoutRequest.account) {
+                // Clear given account.
+                this.browserStorage.removeAccount(AccountEntity.generateAccountCacheKey(logoutRequest.account));
+            } else {
+                // Clear all accounts and tokens
+                this.browserStorage.clear();
+            }
+
+            if (!BrowserUtils.isInIframe()) {
```
Comment (tnorling) [inline lib/msal-browser/src/app/ClientApplication.ts:570]: There was also an ask in #2127 to only clear cache, regardless of iframe or not. Can we provide a flag on the `EndSessionRequest` object to skip the redirect or do something similar to `onRedirectNavigate`?

Comment (jasonnutter) [reply lib/msal-browser/src/app/ClientApplication.ts:570]: Yeah, we can do that.

Comment (jasonnutter) [reply lib/msal-browser/src/app/ClientApplication.ts:570]: The `onRedirectNavigate` PR will add this for logout.

**Pull request closed without merging**